lib/facility.rb

```diff
@@ -1,14 +1,73 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :address,
+              :name,
+              :phone,
+              :services,
+              :collected_fees, 
+              :registered_vehicles,
+              :operating_hours,
+              :holiday_closures
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(office)
+    @address = office[:address]
+    @name = office[:name]
+    @phone = office[:phone]
+    @operating_hours = office[:operating_hours]
+    @holiday_closures = office[:holiday_closures]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return p "Facility does not currently offer this service." unless @services.include?("Vehicle Registration")
+      vehicle.registration_date = Date.today.year
+      @registered_vehicles << vehicle
+      if vehicle.antique?
+        vehicle.plate_type = :antique
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle?
+        vehicle.plate_type = :ev
+        @collected_fees += 200
+      else
+        vehicle.plate_type = :regular
+        @collected_fees += 100
+      end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?("Written Test") && registrant.age >= 16 && registrant.permit?
+      registrant.license_data[:written] = true
+    elsif @services.include?("Written Test") && registrant.age < 16
+      p "Sorry, the registrant is not old enough to take the written test."
+    elsif @services.include?("Written Test") && registrant.permit? == false
+      p "Registrant has not yet earned permit."
+    else
+      p "Facility does not currently offer this service."
+    end
+  end
+
+  def administer_road_test(registrant)
+    if @services.include?("Road Test") && registrant.license_data[:written] == true
```

Comment:
same thing here, whenever we're using a comparison operator and comparing something to a boolean, usually we don't need the comparison, just the first half of it, so this first line could be `if @services .include?("Road Test") && registrant.license_data[:written]`